SPEC.md

```diff
@@ -98,61 +98,114 @@ A build tool is something which reads one or more input artifacts and writes one
+### Gitoid Identifier Types
+
+Git currently supports two varieties of gitoids.  One is based on SHA1 and is in common use.  The other is based on SHA256 and has been very slow to garner adoption.  The [gitoid URI spec](https://www.iana.org/assignments/uri-schemes/prov/gitoid) uses different prefixes,  `gitoid:blob:sha1` or `gitoid:blob:sha256`, to distinguish which algorithm is being used for computing the gitoid of a blob.   This document adopts the gitoid uri prefixes to distinguish Gitoid Identifier Types.  This approach is anticipated to extend gracefully as git adopts new hash types in the future.
+
+All subsequent references to mandatory identifier types in this document should be interpreted to mean the list:
+
+- gitoid:blob:sha1
+- gitoid:blob:sha256
 
 ### GitBOM Document
 
-Each artifact has a GitBOM document that describes its immediate children consiting of a set of new line delimited records, one for each child, in lexical order.
+A GitBOM Document describes the immediate children of an artifact in the Artifact Dependency Graph (ADG).
+
+A GitBOM Document utilizes precisely one Gitoid Identifier Type.
+
+### GitBOM Identifier
+
+A GitBOM Document is identified by computing its identifier as an artifact with the Gitoid Identifier Type used for identifiers within the GitBOM Document itself.
+
+#### GitBOM Document Header
+
+In order to distinguish the type of identifier used in the GitBOM Document, it begins with a single newline terminated header line:
+
+```
+${Gitoid Identifier Type uri prefix}\n
+```
+
+For example:
+
+```
+gitoid:blob:sha1\n
+```
+
+or
+
+```
+gitoid:blob:sha256\n
+```
+
+All identifiers in a GitBOM Document MUST be of the Gitoid Identifier Type declared in the header.
+
+#### GitBOM Document Child Records
+
+The GitBOM Document after the header consists of a list of newline delimited child records
 
 A child artifact which is itself a leaf artifacts would be represented by
 
 ```
-blob⎵${git object id of child}\n
+blob⎵${artifact id of child}\n
 ```
 
 A child artifact which is itself a derived artifact would be represented by
 ```
-blob⎵${git object id of child}⎵bom⎵${git object id of child's GitBOM document}\n
+blob⎵${artifact id of child}⎵bom⎵${GitBOM identifier of child's GitBOM Document}\n
 ```
 
-### GitBOM identifier
+The child artifact records must be written to the GitBOM Document in lexical order.
 
-For a given artifact, the GitBOM ID of that artifact is simply the git object id of it's GitBOM Document.
+The artifact id and GitBOM Document id must both be of the Gitoid Identifier Type declared in the GitBOM Document header.
 
 ### GitBOM Identifier Embedding
 
-Each build tool should embed the GitBOM ID of its output artifact(s) into the output artifact in a manner:
+Each build tool should embed into the output artifact a new line delimited, lexically ordered, list of GitBOM identifiers for each mandatory Gitoid Identifier Type in a manner:
```

Comment:
> Also I think its implied here that the file contents should be treated as binary - ie, the line endings are literal `\n` not whatever the host's line-ending convention is. Likewise all the text is pure ascii, so unicode or other encoding normalization does not come into play. Similarly, "lexically ordered" is over the symbols as bytes, not treating the bytes as text in under any particular locale's ordering.
> 
> I don't remember if the spec already has a section explicitly calling out how these hex strings are encoded (ie fixed length, leading zeros intact, lowercase). But it should have one, and every mention of hex strings should refer to that (rather than leaving it unspecified, or repeating the same text over and over).
> 
> "Delimited" implies the last record does not have a trailing `\n` - is that intended?

I think this has been addressed in #23 .  Let me know what you think.

Reply:
Also I believe I previously shifted from delimited to terminated :)